be/src/vec/aggregate_functions/aggregate_function_multi_topn.h

```diff
@@ -0,0 +1,267 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <rapidjson/encodings.h>
+#include <rapidjson/stringbuffer.h>
+#include <rapidjson/writer.h>
+
+#include <cstdint>
+#include <string>
+
+#include "vec/aggregate_functions/aggregate_function.h"
+#include "vec/aggregate_functions/aggregate_function_multi_top.h"
+#include "vec/columns/column.h"
+#include "vec/columns/column_array.h"
+#include "vec/columns/column_string.h"
+#include "vec/columns/column_struct.h"
+#include "vec/columns/column_vector.h"
+#include "vec/columns/columns_number.h"
+#include "vec/common/assert_cast.h"
+#include "vec/common/space_saving.h"
+#include "vec/common/string_ref.h"
+#include "vec/core/types.h"
+#include "vec/data_types/data_type_array.h"
+#include "vec/data_types/data_type_ipv4.h"
+#include "vec/data_types/data_type_nullable.h"
+#include "vec/data_types/data_type_struct.h"
+#include "vec/io/io_helper.h"
+
+namespace doris::vectorized {
+
+inline constexpr UInt64 TOP_K_MAX_SIZE = 0xFFFFFF;
+
+struct AggregateFunctionTopKGenericData {
+    using Set = SpaceSaving<StringRef, StringRefHash>;
+
+    Set value;
+};
+
+class AggregateFunctionMultiTopN final
+        : public IAggregateFunctionDataHelper<AggregateFunctionTopKGenericData,
+                                              AggregateFunctionMultiTopN>,
+          AggregateFunctionMultiTop {
+private:
+    using State = AggregateFunctionTopKGenericData;
+
+public:
+    AggregateFunctionMultiTopN(const DataTypes& argument_types_)
+            : IAggregateFunctionDataHelper<AggregateFunctionTopKGenericData,
+                                           AggregateFunctionMultiTopN>(argument_types_),
+              column_size(argument_types_.size() - 2) {}
+
+    String get_name() const override { return "multi_topn"; }
```

Comment:
approx_topn is better since the key semantics is approximate and multi filed is not necessary.